### PR TITLE
Adding integrations to the Customer view page

### PR DESCRIPTION
**Pull request by kiram15**
### Description
As part of the enterprise setup support tool update, we created a customer card with relevant customer info.

### UI
Screenshot 2024-08-12 at 1 56 26 PM
Screenshot 2024-08-12 at 1 56 37 PM
Screenshot 2024-08-12 at 1 56 52 PM

### Test Plan on Stage
checkout branch kiram15/ENT-9164 and run npm install
testing link: https://localhost.stage.edx.org:18450/enterprise-configuration/customers/

at the root directory, add file webpack.dev.config.js with the following content:
const { createConfig } = require('@openedx/frontend-build');

module.exports = createConfig('webpack-dev', {
  devServer: {
    allowedHosts: 'all',
    https: true,
  },
});
replace the content in .env.development with this info:
NODE_ENV='development'
PORT=18450
FEATURE_CUSTOMER_SUPPORT_VIEW='true'
ADMIN_PORTAL_BASE_URL='https://portal.stage.edx.org'
ACCESS_TOKEN_COOKIE_NAME='stage-edx-jwt-cookie-header-payload'
BASE_URL='https://localhost.stage.edx.org:18450'
FEATURE_CONFIGURATION_MANAGEMENT='true'
FEATURE_CONFIGURATION_ENTERPRISE_PROVISION='true'
FEATURE_CONFIGURATION_EDIT_ENTERPRISE_PROVISION='true'
CREDENTIALS_BASE_URL='https://credentials.stage.edx.org'
CSRF_TOKEN_API_PATH='/csrf/api/v1/token'
ECOMMERCE_BASE_URL='https://ecommerce.stage.edx.org'
ENTERPRISE_ACCESS_BASE_URL='https://enterprise-access.stage.edx.org'
LANGUAGE_PREFERENCE_COOKIE_NAME='openedx-language-preference'
LMS_BASE_URL='https://courses.stage.edx.org'
LICENSE_MANAGER_URL='https://license-manager.stage.edx.org'
SUPPORT_CONFLUENCE='https://support-tools.edx.org'
SUPPORT_CUSTOMER_REQUEST='https://support-tools.edx.org'
DISCOVERY_API_BASE_URL='https://discovery.stage.edx.org'
LOGIN_URL='https://courses.stage.edx.org/login'
LOGOUT_URL='https://courses.stage.edx.org/logout'
LOGO_URL=https://edx-cdn.org/v3/default/logo.svg
LOGO_TRADEMARK_URL=https://edx-cdn.org/v3/default/logo-trademark.svg
LOGO_WHITE_URL=https://edx-cdn.org/v3/default/logo-white.svg/
FAVICON_URL=https://edx-cdn.org/v3/default/favicon.ico
MARKETING_SITE_BASE_URL='https://stage.edx.org'
ORDER_HISTORY_URL='https://orders.stage.edx.org/orders'
REFRESH_ACCESS_TOKEN_ENDPOINT='https://courses.stage.edx.org/login_refresh'
SEGMENT_KEY=null
SITE_NAME='edX'
SUBSIDY_BASE_URL='https://enterprise-subsidy.stage.edx.org'
USER_INFO_COOKIE_NAME='edx-user-info'
PUBLISHER_BASE_URL='https://publisher.stage.edx.org/'
APP_ID='support-tools'
MFE_CONFIG_API_URL='https://courses.stage.edx.org/api/mfe_config/v1'

verify that you can see the integration cards match figma: https://www.figma.com/design/TP5hGfoYmohr21MwgUgCZN/Enterprise-Setup?node-id=914-16353&t=3Fs3hMg0TNnSl7Lv-0
Verify with multiple customers that all 3 cards (API, SSO, and LMS) are all working and showing up. 

### Jira
https://2u-internal.atlassian.net/browse/ENT-9164

https://github.com/user-attachments/assets/2b4529aa-0be5-42a0-8ef4-1998375057e0

### CODE REVIEW

File: src/Configuration/Customers/CustomerDetailView/CustomerIntegrations.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            title={sso.displayName}
+            subtext={ssoDateText(sso)}
+            buttonText="Open in Admin Portal"
+            buttonLink={`${ADMIN_PORTAL_BASE_URL}/${slug}/admin/settings/sso`}
+          />
+        ))}
+        {activeIntegrations && activeIntegrations.map((config) => (
+          <CustomerViewCard
+            slug={slug}
+            header="Learner platform"
```
Comment (marlonkeating) [inline src/Configuration/Customers/CustomerDetailView/CustomerIntegrations.jsx]: Following verbiage in figma
```suggestion
            header="Learning platform"
```

File: src/Configuration/Customers/CustomerDetailView/CustomerIntegrations.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import CustomerViewCard from './CustomerViewCard';
+import { formatDate } from '../data/utils';
+import DJANGO_ADMIN_BASE_URL from '../data/constants';
+
+const CustomerIntegrations = ({
+  slug, activeIntegrations, activeSSO, apiCredentialsEnabled,
+}) => {
+  const { ADMIN_PORTAL_BASE_URL } = getConfig();
+  const ssoDateText = ({ sso }) => (`Created ${formatDate(sso?.created)} • Last modified ${formatDate(sso?.modifed)}`);
+  const configDateText = ({ config }) => (`Last modified ${formatDate(config?.lastModifiedAt)}`);
```
Comment (marlonkeating) [inline src/Configuration/Customers/CustomerDetailView/CustomerIntegrations.jsx]: Figma design calls for created date on lms entries too.  Is that to be implemented still, or do we not have access to that data period?

File: src/Configuration/Customers/data/constants.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+const DJANGO_ADMIN_BASE_URL = 'https://internal.courses.edx.org';
```
Comment (marlonkeating) [inline src/Configuration/Customers/data/constants.js:1]: Can this value can be configured in .env?

**Review by marlonkeating — CHANGES_REQUESTED** [review]

File: src/Configuration/Customers/CustomerDetailView/CustomerIntegrations.jsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  return (
+    <Container className="mt-3 pr-6 mb-5">
+      <div>
+        <h2 className="pt-4">Associated Integrations</h2>
```
Comment (marlonkeating) [inline src/Configuration/Customers/CustomerDetailView/CustomerIntegrations.jsx:20]: Should we hide this when no integrations are present?  That's what the designs seem to indicate.

File: src/Configuration/Customers/data/constants.js (hunk 3 above)
Comment (kiram15) [reply src/Configuration/Customers/data/constants.js:1]: Usually we're pulling from `import { getConfig } from '@edx/frontend-platform'` so I was talking to Katrina about maybe putting in a ticket to add it in there?

**Review by marlonkeating — APPROVED** [review]

**Pull request merged**